server/datastore/mysql/migrations/tables/20220714191431_AddHostOperatingSystemTable.go

```diff
@@ -0,0 +1,29 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220714191431, Down_20220714191431)
+}
+
+func Up_20220714191431(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE host_operating_system (
+    host_id INT UNSIGNED NOT NULL PRIMARY KEY,
+    os_id INT UNSIGNED NOT NULL
```

Comment:
Consider adding an index here. Also, I think it might be 'ok' to add a foreign key - we will be doing way more reads than writes/updates to the `operating_systems` table, I don't think we will be running into locking issues if we add a FK.

Reply:
(I've just re-read https://github.com/fleetdm/fleet/tree/main/handbook/engineering#foreign-keys-and-locking.)

So, not sure about the reads vs writes wrt these new tables (by default, hosts check in with new details data every ~1h). But IMO it makes sense to define a foreign key to `operating_systems` here.

Reply:
+1 to index. Not sure how I feel about the FK, tbh, been burned so much by it that I would avoid it, but it might be an overkill as you say.